test_unstructured/chunking/test_title.py

```diff
@@ -22,6 +22,100 @@
 )
 from unstructured.partition.html import partition_html
 
+# == chunk_by_title() validation behaviors =======================================================
+
+
+@pytest.mark.parametrize("max_characters", [0, -1, -42])
+def test_it_rejects_max_characters_not_greater_than_zero(max_characters: int):
+    elements: List[Element] = [Text("Lorem ipsum dolor.")]
+
+    with pytest.raises(
+        ValueError, match=f"'max_characters' argument must be > 0, got {max_characters}"
+    ):
+        chunk_by_title(elements, max_characters=max_characters)
+
+
+def test_it_does_not_complain_when_specifying_max_characters_by_itself():
+    """Caller can specify `max_characters` arg without specifying any others.
+
+    In particular, When `combine_text_under_n_chars` is not specified it defaults to the value of
+    `max_characters`; it has no fixed default value that can be greater than `max_characters` and
+    trigger an exception.
+    """
+    elements: List[Element] = [Text("Lorem ipsum dolor.")]
+
+    try:
+        chunk_by_title(elements, max_characters=50)
+    except Exception:
+        pytest.fail("did not accept `max_characters` as option by itself")
+
+
+@pytest.mark.parametrize("n_chars", [-1, -42])
+def test_it_rejects_combine_text_under_n_chars_for_n_less_than_zero(n_chars: int):
+    elements: List[Element] = [Text("Lorem ipsum dolor.")]
+
+    with pytest.raises(
+        ValueError, match=f"'combine_text_under_n_chars' argument must be >= 0, got {n_chars}"
+    ):
+        chunk_by_title(elements, combine_text_under_n_chars=n_chars)
+
+
+def test_it_accepts_0_for_combine_text_under_n_chars_to_disable_chunk_combining():
+    """Specifying `combine_text_under_n_chars=0` is how a caller disables chunk-combining."""
+    elements: List[Element] = [Text("Lorem ipsum dolor.")]
+
+    chunks = chunk_by_title(elements, max_characters=50, combine_text_under_n_chars=0)
+
+    assert chunks == [CompositeElement("Lorem ipsum dolor.")]
+
+
+@pytest.mark.parametrize("n_chars", [-1, -42])
+def test_it_rejects_new_after_n_chars_for_n_less_than_zero(n_chars: int):
+    elements: List[Element] = [Text("Lorem ipsum dolor.")]
+
+    with pytest.raises(
+        ValueError, match=f"'new_after_n_chars' argument must be >= 0, got {n_chars}"
+    ):
+        chunk_by_title(elements, new_after_n_chars=n_chars)
+
+
+def test_it_does_not_complain_when_specifying_new_after_n_chars_by_itself():
```

Comment:
Should `test_it_does_not_complain_when_specifying_combine_text_under_n_chars_by_itself` also exist (I see it exists here and for max_characters)? If not, what other param is required with it, and can we test that it warns/errs as needed?